src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -79,13 +80,20 @@ RebuildQueryStrings(Job *workerJob)
 
 		if (UpdateOrDeleteQuery(query))
 		{
+			List *relationShardList = task->relationShardList;
+
 			/*
 			 * For UPDATE and DELETE queries, we may have subqueries and joins, so
 			 * we use relation shard list to update shard names and call
 			 * pg_get_query_def() directly.
 			 */
-			List *relationShardList = task->relationShardList;
 			UpdateRelationToShardNames((Node *) query, relationShardList);
+
+			/*
+			 * For UPDATE and DELETE queries, we have relations wrapped in citus container
+			 * function. We traverse those, and convert them to shards.
+			 */
+			UpdateCitusContainerWrappedRelationToShardNames(query, relationShardList);
```

Comment:
It's a bit strange that we do this right after UpdateRelationToShardNames, because we basically transform the RTEs again.

Could we integrate this into UpdateRelationToShardNames?